generator/internal/http_option_utils.cc

```diff
@@ -242,21 +252,37 @@ ParseHttpExtension(google::protobuf::MethodDescriptor const& method) {
                      << ": google::api::HttpRule not handled";
   }
 
-  static std::regex const kUrlPatternRegex(R"((.*)\{(.*)=(.*)\}(.*))");
-  std::smatch match;
-  if (!std::regex_match(url_pattern, match, kUrlPatternRegex)) {
+  static std::regex const kImplicitNamedFieldUrlPatternRegex(
+      R"((.*)\{(.*)\}(.*))");
```

Comment:
Can you have something like: `"/v1/{parent=projects/*/instances/*}/databases/{bar}/baz"`?

Is there a full grammar for these expressions we are trying to match?

Reply:
I found it:

https://github.com/googleapis/googleapis/blob/02aeb715005c633115bc77d563d6920731c98189/google/api/http.proto#L227-L234